sympy/functions/combinatorial/numbers.py

```diff
@@ -1400,6 +1401,181 @@ def _eval_is_positive(self):
         if n.is_nonnegative and n.is_integer:
             return True
 
+#----------------------------------------------------------------------------#
+#                                                                            #
+#                             Motzkin numbers                                #
+#                                                                            #
+#----------------------------------------------------------------------------#
+
+
+class motzkin(Function):
+    r"""
+    Motzkin numbers
+
+    Motzkin number is the number of different ways of drawing
+    non-intersecting chords between n points on a circle
+    (not necessarily touching every point by a chord).
+    The numbers satisfy the recurrence relation
+
+    .. math :: M_n = \frac{2n + 1}{n + 2} M_{n - 1} + \frac{3n - 3}{n + 2} M_{n - 2}
+
+    For non-negative integers, it can be calculated as
+
+    .. math :: M_n = \sum_0^{\left\lfloor n/2 \right\rfloor } {n \choose 2k} C_k
+
+    where `C_k` is the `k^{th}` Catalan number.
+
+    It can be generalized to complex numbers as
+
+    .. math :: M_n = {}_2 F_1\Bigg(\frac{\frac{1 - n}{2}, \frac{-1}{2}}{2} \Bigg{|} 4\Bigg)
+
+    where `{}_p F_q` is the Generalized Hypergeometric function.
+
+    * ``motzkin(n)`` gives the `n^{th}` Motzkin number.
+
+    * ``find_first_n_motzkin(n)`` gives the first
```

Comment:
I think most of these don't need to be there (see the comments below), but for the ones that stay, they should be documented on the method itself.